Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -460,9 +464,88 @@
+        return 0;
+    case PDFDocumentLayout::DisplayMode::Continuous: {
+        // Walk down the document, find the first page bound that contains the position.
+        // Note that we don't consider display direction (vertical vs horizontal).
+        for (PDFDocumentLayout::PageIndex index = 0; index < m_documentLayout.pageCount(); ++index) {
+            auto pageBounds = m_documentLayout.boundsForPageAtIndex(index);
+            pageBounds.expand(PDFDocumentLayout::documentMargin);
+            pageBounds.contract(PDFDocumentLayout::pageMargin);
+            // At this point, this page index necessarily corresponds to the
+            // nearest page to the point (so far), so we always keep track of it.
+            nearestPageIndex = index;
+            if (point.y() <= pageBounds.maxY())
+                break;
+        }
+    }
+    }
+
+    return nearestPageIndex;
+}
+
+WebCore::IntPoint UnifiedPDFPlugin::pointInPDFPageSpaceForDocumentPoint(const WebCore::IntPoint& point, PDFDocumentLayout::PageIndex pageIndex) const
+{
+    ASSERT(pageIndex < m_documentLayout.pageCount());
+
+    auto pageBounds = m_documentLayout.boundsForPageAtIndex(pageIndex);
+    auto pageRotation = m_documentLayout.rotationForPageAtIndex(pageIndex);
+    auto pointInPDFPageSpace = IntPoint { point - WebCore::flooredIntPoint(pageBounds.location()) };
+    auto documentSpaceToPageSpaceTransform = AffineTransform::makeRotation(pageRotation).translate([&pageBounds, pageRotation] () -> FloatPoint {
+        auto width = pageBounds.width();
+        auto height = pageBounds.height();
+        switch (pageRotation) {
+        case 0:
+            return { };
+        case 90:
+            return { 0, height };
+        case 180:
+            return { height, width };
+        case 270:
+            return { width, 0 };
+        default:
+            ASSERT_NOT_REACHED();
+            return { };
+        }
+    }());
+
+    auto pointWithTopLeftOrigin = documentSpaceToPageSpaceTransform.mapPoint(pointInPDFPageSpace);
+    return IntPoint { pointWithTopLeftOrigin.x(), static_cast<int>(pageBounds.height()) - pointWithTopLeftOrigin.y() };
+}
+
+auto UnifiedPDFPlugin::pdfElementTypesForPluginPoint(const WebCore::IntPoint& point) const -> PDFElementTypes
+{
+    auto pointInDocumentSpace = pointInDocumentSpaceForPluginPoint(point);
+    auto maybeNearestPageIndex = nearestPageIndexForDocumentPoint(pointInDocumentSpace);
+    if (!maybeNearestPageIndex || *maybeNearestPageIndex >= m_documentLayout.pageCount())
+        return { };
+    auto nearestPageIndex = *maybeNearestPageIndex;
+    auto pointInPDFPageSpace = pointInPDFPageSpaceForDocumentPoint(pointInDocumentSpace, nearestPageIndex);
+
+    if (auto annotation = [m_documentLayout.pageAtIndex(nearestPageIndex) annotationAtPoint:pointInPDFPageSpace]) {
+        // FIXME: Reason about the annotation type to construct the full set of PDFElementTypes at this page point.
+        (void)annotation;
```

Comment:
UNUSED_PARAM, perhaps

Reply:
Fixed in [af10ddc](https://github.com/WebKit/WebKit/commit/af10ddcd0e5fdcb907e4c6dc388e54411872b172)